app/src/test/java/nl/tudelft/b_b_w/controller/KeyStorageUnitTest.java

```diff
@@ -95,15 +95,10 @@ public void testReadKeyEmpty() throws IOException {
      * Test to check whether converting a public key to a string and back works
      */
     @Test
-    public void testStringPublicKey() {
-        try {
-            EdDSAPublicKey edDSAPublicKey1 = KeyReader.readPublicKey(
-                    KeyWriter.publicKeyToString(edDSAPublicKey));
-            assertEquals(edDSAPublicKey, edDSAPublicKey1);
-        } catch (InvalidKeySpecException e) {
-            e.printStackTrace();
-            fail();
-        }
+    public void testStringPublicKey()  throws Exception {
```

Comment:
Do not address code that is not in your issue. Since you modified the code in a dirty way (added throws exception), it will cause merge conflicts and will have to be edited later. Next to that, you did not specify any reason, why this should be changed.